[2025-04-18 Fiddler](https://thefiddler.substack.com/p/can-you-throw-the-hammer)
====================

I'll assume reaching 4 points on a 2-point hole also wins.

I'm having a hard time imagining when it would be advantageous
to reject the hammer.  If you are up 2-0, it might be, but it's
not obvious without working through the probabilities.

Fiddler
-------
If the score is 2-2, either player has a 50% chance of winning,
irrespective of the hammer.

If the score is 2-1, the losing player will never reject the
hammer.  If the winning player rejects the hammer, the score
will be 2-2, otherwise either player has a 50% chance of winning.
If no hammer is thrown, the winning player will have a 75%
chance of winning, so the losing player will always throw the
hammer.  So a 2-1 score means either player has a 50% chance of
winning.

If the score is 2-0, the losing player will never reject the
hammer.  If the winning player rejects the hammer, it will be
2-1 and each player has a 50% chance of winning, so the winning
player will not reject the hammer, since accepting the hammer
means having a 75% chance of winning.  If no hammer is thrown,
the winning player will have a 75% chance of winning.  So a
2-0 score means the winning player has a 75% chance of winning.

If the score is 1-1, a rejected hammer means going to 2-1 and
a 50% chance of winning.  An accepted hammer means a 50% chance
of winning.  No hammer means going to 2-1 and a 50% chance of
winning.  So a 1-1 score means a 50% chance of winning.

Finally, if the score is 1-0 and no hammer is thrown, the winning
player has 5/8 chance of winning.  If the winning player rejects
the hammer, it's 50%.  If the losing player rejects the hammer,
that means 25%.  If the hammer is accepted, the winning player
has 75% chance of winning.  That means the winning player will
throw the hammer and have a 75% chance of winning irrespective
of whether the losing player accepts the hammer.

Extra credit
------------
4-4 means a 50% chance of winning.

4-3 means the losing player will throw the hammer for a 50% chance of winning.

At 4-2, if no hammer is thrown, the winning player has a 75% chance of winning.
If the hammer is accepted, the winning player has a 75% chance of winning.  The
losing player will not reject the hammer for a 0% chance of winning.  The winning
player will not reject the hammer for a 50% chance of winning.

At 4-1, if no hammer is thrown, the winning player has a 7/8 chance of winning.
If the hammer is accepted, the winning player has a 75% chance of winning.  The
losing player will not reject the hammer for a 0% chance of winning.  If the winning
player rejects the hammer, it'll be a 75% chance of winning.  So the losing player
will always throw the hammer for a 25% chance of winning.

At 4-0, if no hammer is thrown, the winning player has a 7/8 chance of winning.
If the hammer is accepted, the winning player has 7/8 chance of winning.  The
losing player will not reject the hammer for 0%, and the winning player will not
reject the hammer for 75%.

3-3 means a 50% chance of winning.

At 3-2, if no hammer is thrown, the winning player has a 7/8 chance of winning.
If the hammer is accepted, the winning player has a 75% chance of winning.
The winning player will not reject the hammer for 50% from 3-3.  If the losing player
rejects the hammer, the winning player has a 75% chance of winning from 4-2.  So
the losing player will always throw the hammer for a 25% chance of winning.

At 3-1, if no hammer is thrown, the winning player has a 75% chance of winning.
If the hammer is accepted, the winning player has a 75% chance of winning.
The winning player can reject the hammer for a 75% chance of winning from 3-2.  The
losing player can reject the hammer for a 25% chance of winning from 4-1.

At 3-0, if no hammer is thrown, the winning player has a 13/16 chance of winning.
If the hammer is accepted, the winning player has a 7/8 chance of winning.  The losing
player can reject the hammer for a 1/8 chance of winning from 4-0.  So the winning
player will always throw the hammer for a 7/8 chance of winning.

2-2 means a 50% chance of winning.

At 2-1, if no hammer is thrown, the winning player has a 5/8 chance of winning.
If the hammer is accepted, the winning player has a 50% chance of winning.
The winning player can reject the hammer for 50% from 2-2.  The losing player
will never reject the hammer for 25% from 3-1.  So the losing player will always
throw the hammer for 50%.

At 2-0, if no hammer is thrown, the winning player has 11/16.  If the hammer is
accepted, the winning player has 11/16.  The winning player will not reject
the hammer for 50% from 2-1.  The losing player will not reject the hammer for
3/16 from 3-0.

1-1 means 50%.

At 1-0, if no hammer is thrown, the winning player has 19/32.  If the hammer is
accepted, the winning player has 11/16.  The winning player will not reject
the hammer for 50% from 1-1.  The losing player can reject the hammer for 5/16
from 2-0.  So the winning player will always throw the hammer for for 11/16.

To check for errors, tabulate the probabilities that the winning player wins:

|score|no hammer              |hammer                 |reject|reject|summary|
|-----|-----------------------|-----------------------|------|------|-------|
|4-4  |1/2                    |1/2                    |0     |1     |1/2    |
|4-3  |1/2⋅1+1/2⋅1/2=3/4      |1/2                    |1/2   |1     |1/2    |
|4-2  |1/2⋅1+1/2⋅1/2=3/4      |1/2⋅1+1/2⋅1/2=3/4      |1/2   |1     |3/4    |
|4-1  |1/2⋅1+1/2⋅3/4=7/8      |1/2⋅1+1/2⋅1/2=3/4      |3/4   |1     |3/4    |
|4-0  |1/2⋅1+1/2⋅3/4=7/8      |1/2⋅1+1/2⋅3/4=7/8      |3/4   |1     |7/8    |
|3-3  |1/2                    |1/2                    |1/2   |1/2   |1/2    |
|3-2  |1/2⋅3/4+1/2⋅1/2=5/8    |1/2⋅1+1/2⋅1/2=3/4      |1/2   |3/4   |3/4    |
|3-1  |1/2⋅3/4+1/2⋅3/4=3/4    |1/2⋅1+1/2⋅1/2=3/4      |3/4   |3/4   |3/4    |
|3-0  |1/2⋅7/8+1/2⋅3/4=13/16  |1/2⋅1+1/2⋅3/4=7/8      |3/4   |7/8   |7/8    |
|2-2  |1/2                    |1/2                    |3/4   |3/4   |1/2    |
|2-1  |1/2⋅3/4+1/2⋅1/2=7/8    |1/2⋅3/4+1/2⋅1/4=1/2    |1/2   |3/4   |1/2    |
|2-0  |1/2⋅7/8+1/2⋅1/2=11/16  |1/2⋅7/8+1/2⋅1/2=11/16  |1/2   |7/8   |11/16  |
|1-1  |1/2                    |1/2                    |1/2   |1/2   |1/2    |
|1-0  |1/2⋅11/16+1/2⋅1/2=19/32|1/2⋅7/8+1/2⋅1/2=11/16  |1/2   |11/16 |11/16  |

And it looks like there is no case for which it's better for both players to not
throw the hammer, and there is no case where rejecting the hammer is better than
accepting the hammer.

Making the $\lfloor$rounds$\rceil$
----------------------------------
The only pieces that need special tracking are the bishops, the rooks, and the king.  That's
most of the pieces.  The queens and the knights could be tracked separately, or they could be
tracked together then multiply the result by 3 for the different possible permutations.  I'll
track them together.

Consider placing the pieces from left to right.  Once the first bishop is placed, the second
bishop is restricted.  The king cannot be placed until after the first rook is placed and the
second rook cannot be placed until the king is placed.

There are four possible bishop states: neither placed, one placed on white, one placed on black,
and both placed.  There are four possible rook/king states: none placed, one rook placed, king
placed, and all three placed.

Let the left-most square be black.  The state is initially B0,K0,Q0.  There are 3 distinct pieces
that can be placed on the left-most square.  If it is a bishop, the state becomes BB,K0,Q0.  If
it is a rook, the state becomes B0,K1,Q0.  If it is a queen or knight, the state becomes B0,K0,Q1.

Then, the second piece will be placed on white.  The possible states from BB,K0,Q0 are B2,K0,Q0,
BB,K1,Q0, and BB,K0,Q1.  The possible states from B0,K1,Q0 are BW,K1,Q0, B0,K2,Q0, and B0,K1,Q1.
The possible states from B0,K0,Q1 are BW,K0,Q1, B0,K1,Q1, B0,K0,Q2.

Using code:

In [1]:
from functools import cache

@cache
def count_arrangements(i, b, k, q):
    if i > 8:
        return 1
    n = 0
    if b < 3:
        if b == 0:
            n += count_arrangements(i+1, 1 if is_even(i) else 2, k, q)
        elif b != (1 if is_even(i) else 2):
            n += count_arrangements(i+1, 3, k, q)
    if k < 3:
        n += count_arrangements(i+1, b, k+1, q)
    if q < 3:
        n += count_arrangements(i+1, b, k, q+1)
    return n

count_arrangements(1, 0, 0, 0)*3

960

960 possible starting positions is small enough to list them all.

In [2]:
def list_arrangements(i, arr, b, k, q, n):
    if i > 8:
        print(arr)
        return
    if b < 3:
        if b == 0:
            list_arrangements(i+1, arr+"B", 1 if is_even(i) else 2, k, q, n)
        elif b != (1 if is_even(i) else 2):
            list_arrangements(i+1, arr+"B", 3, k, q, n)
    if k == 0:
        list_arrangements(i+1, arr+"R", b, 1, q, n)
    elif k == 1:
        list_arrangements(i+1, arr+"K", b, 2, q, n)
    elif k == 2:
        list_arrangements(i+1, arr+"R", b, 3, q, n)
    if q == 0:
        list_arrangements(i+1, arr+"Q", b, k, 1, n)
    if n < 2:
        list_arrangements(i+1, arr+"N", b, k, q, n+1)

list_arrangements(1, "", 0, 0, 0, 0)

BBRKRQNN
BBRKRNQN
BBRKRNNQ
BBRKQRNN
BBRKQNRN
BBRKQNNR
BBRKNRQN
BBRKNRNQ
BBRKNQRN
BBRKNQNR
BBRKNNRQ
BBRKNNQR
BBRQKRNN
BBRQKNRN
BBRQKNNR
BBRQNKRN
BBRQNKNR
BBRQNNKR
BBRNKRQN
BBRNKRNQ
BBRNKQRN
BBRNKQNR
BBRNKNRQ
BBRNKNQR
BBRNQKRN
BBRNQKNR
BBRNQNKR
BBRNNKRQ
BBRNNKQR
BBRNNQKR
BBQRKRNN
BBQRKNRN
BBQRKNNR
BBQRNKRN
BBQRNKNR
BBQRNNKR
BBQNRKRN
BBQNRKNR
BBQNRNKR
BBQNNRKR
BBNRKRQN
BBNRKRNQ
BBNRKQRN
BBNRKQNR
BBNRKNRQ
BBNRKNQR
BBNRQKRN
BBNRQKNR
BBNRQNKR
BBNRNKRQ
BBNRNKQR
BBNRNQKR
BBNQRKRN
BBNQRKNR
BBNQRNKR
BBNQNRKR
BBNNRKRQ
BBNNRKQR
BBNNRQKR
BBNNQRKR
BRKBRQNN
BRKBRNQN
BRKBRNNQ
BRKBQRNN
BRKBQNRN
BRKBQNNR
BRKBNRQN
BRKBNRNQ
BRKBNQRN
BRKBNQNR
BRKBNNRQ
BRKBNNQR
BRKRQBNN
BRKRQNNB
BRKRNBQN
BRKRNBNQ
BRKRNQNB
BRKRNNQB
BRKQRBNN
BRKQRNNB
BRKQNBRN
BRKQNBNR
BRKQNRNB
BRKQNNRB
BRKNRBQN
BRKNRBNQ
BRKNRQNB
BRKNRNQB
BRKNQBRN
BRKNQBNR
BRKNQRNB
BRKNQNRB
BRKNNBRQ
BRKNNBQR
BRKNNRQB
BRKNNQRB
BRQBKRNN
BRQBKNRN
BRQBKNNR
BRQBNKRN
BRQBNKNR
BRQBNNKR
BRQKRBNN
BRQKRNNB
BRQKNBRN
BRQKNBNR
BRQKNRNB
BRQKNNRB
BRQNKBRN
BRQNKBNR
BRQNKRNB
B